### PR TITLE
rpcclient: fix filtered naive subscriptions receiver

### LINKED ISSUE #2835: Panic on send to naive WS Client's receiver with filter
```
panic: interface conversion: interface {} is *neorpc.BlockFilter, not neorpc.BlockFilter

goroutine 61 [running]:
[github.com/nspcc-dev/neo-go/pkg/neorpc/rpcevent.Matches({0x1304608,](http://github.com/nspcc-dev/neo-go/pkg/neorpc/rpcevent.Matches(%7B0x1304608,) 0xc0002e0120}, {0x13089b0, 0xc0002d3f38})
        /home/kira/work/neofs-cdn-edge-manager/vendor/[github.com/nspcc-dev/neo-go/pkg/neorpc/rpcevent/filter.go:38](http://github.com/nspcc-dev/neo-go/pkg/neorpc/rpcevent/filter.go:38) +0xd05
[github.com/nspcc-dev/neo-go/pkg/rpcclient.(*naiveReceiver).TrySend(0xc0002e0120,](http://github.com/nspcc-dev/neo-go/pkg/rpcclient.(*naiveReceiver).TrySend(0xc0002e0120,) {0x1, {0x1151de0, 0xc0002e2d80}})
        /home/kira/work/neofs-cdn-edge-manager/vendor/[github.com/nspcc-dev/neo-go/pkg/rpcclient/wsclient.go:296](http://github.com/nspcc-dev/neo-go/pkg/rpcclient/wsclient.go:296) +0x8e
[github.com/nspcc-dev/neo-go/pkg/rpcclient.(*WSClient).wsReader(0xc00025b680)](http://github.com/nspcc-dev/neo-go/pkg/rpcclient.(*WSClient).wsReader(0xc00025b680))
        /home/kira/work/neofs-cdn-edge-manager/vendor/[github.com/nspcc-dev/neo-go/pkg/rpcclient/wsclient.go:494](http://github.com/nspcc-dev/neo-go/pkg/rpcclient/wsclient.go:494) +0x1462
created by [github.com/nspcc-dev/neo-go/pkg/rpcclient.NewWS](http://github.com/nspcc-dev/neo-go/pkg/rpcclient.NewWS)
        /home/kira/work/neofs-cdn-edge-manager/vendor/[github.com/nspcc-dev/neo-go/pkg/rpcclient/wsclient.go:390](http://github.com/nspcc-dev/neo-go/pkg/rpcclient/wsclient.go:390) +0x7c5
```
Thanks to @masterSplinter01 for discovering. Could you please, specify the NeoGo version for clarification?

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Close #2835.

### CODE REVIEW

File: pkg/rpcclient/wsclient.go [hunk 1]
```diff
@@ -283,7 +283,33 @@ func (r *naiveReceiver) EventID() neorpc.EventID {
 
 // Filter implements neorpc.Comparator interface.
 func (r *naiveReceiver) Filter() interface{} {
-	return r.filter
+	if r.filter == nil {
+		return nil
+	}
+	switch f := r.filter.(type) {
+	case *neorpc.BlockFilter:
+		if f == (*neorpc.BlockFilter)(nil) {
```
Comment (roman-khimov) [inline pkg/rpcclient/wsclient.go:291]: Do you need a type cast here? `f` already has an appropriate type.

Comment (AnnaShaleva) [reply pkg/rpcclient/wsclient.go]: This was added due to the way how `r.filter` was initialized (https://github.com/nspcc-dev/neo-go/pull/2836#pullrequestreview-1208375867), now this code is removed.

**Comment by AnnaShaleva** [comment]
> Why don't you try something like this?
> 
> ```
> func (c *WSClient) SubscribeForNewBlocks(primary *int) (string, error) {
> -        var flt *neorpc.BlockFilter
> +        var flt interface{}
> ```
> 
> Seems to be somewhat simpler and should work as well.

Just didn't come to my mind, fixed.

**Comment by AnnaShaleva** [comment]
Oh, linter is unhapy.

File: pkg/rpcclient/wsclient.go [hunk 2]
```diff
@@ -640,13 +654,13 @@ func (c *WSClient) performSubscription(params []interface{}, rcvr notificationRe
 //
 // Deprecated: please, use ReceiveBlocks. This method will be removed in future versions.
 func (c *WSClient) SubscribeForNewBlocks(primary *int) (string, error) {
-	var flt *neorpc.BlockFilter
+	var flt interface{}
 	if primary != nil {
 		flt = &neorpc.BlockFilter{Primary: primary}
```
Comment (roman-khimov) [inline pkg/rpcclient/wsclient.go:659]: And if you're to do `neorpc.BlockFilter{Primary: primary}` here, you can avoid changing `Filter()` at all.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**